Remove `check` from `all` target in Makefile

Makefile currently contains a `check` target under the `all` target, which doesn't actually exist. This makes `make` fail with

```console
$ make
make: *** No rule to make target 'check', needed by 'all'.  Stop.
```

https://github.com/celestiaorg/celestia-core/blob/f4cf3df6e222160a1c4a3c66841dbd0ec17eb408/Makefile#L58

--- a/Makefile
+++ b/Makefile
@@ -55,7 +55,7 @@ endif
 # allow users to pass additional flags via the conventional LDFLAGS variable
 LD_FLAGS += $(LDFLAGS)
 
-all: check build test install
+all: build test install
 .PHONY: all
 
 include tests.mk

